Adding WANDB example (#199)
mila-iqia/mila-docs: docs/examples/good_practices/wandb_setup/index.rst

WANDB Setup
=====================================

Prerequisites:

* :doc:`/examples/frameworks/pytorch_setup/index`
* :doc:`/examples/distributed/single_gpu/index`

Make sure to create a WANDB account, then you can either :
- Set your WANDB_API_KEY environment variable
- Run `wandb login` from the command line

* `<https://wandb.ai/site>`_

Other interesting resources:

* `<https://docs.wandb.ai/quickstart>`_

Click here to see `the source code for this example

> <https://github.com/mila-iqia/mila-docs/tree/master/docs/examples/good_practices/wandb_setup>`_

Reviewer: Hello ! Je crois que ce lien mène vers une page Github 404 (not found).

Si nécessaire, on peut voir le rendu de cet exemple ici (le lien est disponible dans les résultats CI de chaque commit de la PR): https://mila-docs--199.org.readthedocs.build/en/199/examples/good_practices/wandb_setup/index.html#

Reviewer: Oh, ok ! Donc c'est vraiment un lien direct vers Github, pas un lien vers une section de la doc.

Dans ce cas, c'est bon pour moi ! @lebrice @gyom

**job.sh**

.. literalinclude:: job.sh.diff
    :language: diff

**main.py**

.. literalinclude:: main.py.diff
    :language: diff

**Running this example**

> Note : On DRAC clusters you will need to run ``wandb off`` to log your data as offline mode.
> You will then be able to upload your runs with the command ``wandb sync --sync-all``

Reviewer 2: ```suggestion
Note : On DRAC clusters you will need to run `module load httpproxy` to allow
access to wandb servers and log your data.
```

Reviewer 3: Selon @bouthilx  c'est une mauvaise idée de parler de `httpproxy` car les jobs peuvent hang à la fin

Reviewer 2: Ah ok merci!

WANDB Setup
=====================================

Prerequisites:

* :doc:`/examples/frameworks/pytorch_setup/`
* :doc:`/examples/distributed/single_gpu/`

Make sure to create a WANDB account, then you can either :
- Set your WANDB_API_KEY environment variable
- Run `wandb login` from the command line

* `<https://wandb.ai/site>`_

You should also set your `WANDB_HOME` somewhere in `$SCRATCH`, which you can do by adding a command like

> `export WANDB_HOME=$SCRATCH/wandb` to your `~/.bash_aliase` file.

Author: Après avoir testé, `WANDB_HOME` semble rien faire. C'était pour les logs qu'on voulait utiliser ça ?
